### DISCUSSION TITLE
Add Error handling block as available in other languages like Delphi

### DISCUSSION THREAD

**Discussion by YogiYang** [twinbasic/twinbasic#731] [Ideas]
If this has been stated I could not find it so I am suggesting it here.

Besides giving Error handling as per VB6 also add support for Try...Finally and Try...Exception blocks.

Such error handling is very common in other programming languages like Delphi, Pascal, PHP, etc.

Personally I feel that adding this feature will also add a modern touch to BASIC as implemented in tB.

**Comment by WaynePhillipsEA** [comment]
vbWatchdog will soon be coming to twinBASIC.  With that, you get support for familiar `Catch` / `CatchAll` / `Finally` _blocks***_, in a more VBx-friendly way:

https://www.everythingaccess.com/vbwatchdog/bookmarks.htm

```vb
Public Sub SimulateAnError()

    Debug.Print 1 / 0	' Simulate division by zero error (error 11)
    Debug.Print "A" / 1 ' Simulate type mismatch error (error 13)
    Err.Raise &H123, , "This should be caught by the catch &H123 block..."
    Err.Raise &H456, , "This should be caught in the catch-all block..."
    
    ' Program flow now transfers to Finally...
        
ErrEx.Catch 11, 13 ' you can set up a constant/enumeration if you prefer
    MsgBox "Error: either division by zero, or type mismatch!"
    Resume Next

ErrEx.Catch &H123 ' you can set up a constant/enumeration if you prefer
    MsgBox "Error: caught error 291 (&H123)!"
    Resume Next
            
ErrEx.CatchAll     
    MsgBox "Error (catch-all):" & vbCrLf & vbCrLf & Err.Description
    Resume Next
    
ErrEx.Finally
    MsgBox "Finally"
    Debug.Print 1 / 0    ' Errors here automatically ignored due to implicit OnErrorResumeNext
    
End Sub
```

Note that there is no need for `On Error ...` statements; it's all handled implicitly when you add a `Catch` or `CatchAll` block to a procedure.

***Real block scoping is to be discussed in https://github.com/WaynePhillipsEA/twinbasic/issues/436
